geomstats/geometry/matrices.py

```diff
@@ -36,13 +39,36 @@ def mul(a, b):
 
         Returns
         -------
-        c : array-like, shape=[n_samples, dim1, dim3]
+        eq : array-like boolean, shape=[n_samples]
         """
-        return gs.matmul(a, b)
+        is_vectorized = \
+            (gs.ndim(gs.array(a)) == 3) or (gs.ndim(gs.array(b)) == 3)
+        axes = (1, 2) if is_vectorized else (0, 1)
+        return gs.all(gs.isclose(a, b, atol=atol), axes)
 
     @staticmethod
-    def commutator(a, b):
-        """Return the commutator of a and b, i.e. :math: `[a, b] = ab - ba`.
+    def mul(*args):
+        """
+        Return the product of matrices a1, ..., an.
+
+        Parameters
+        ----------
+        a1 : array-like, shape=[n_samples, dim_1, dim_2]
+        a2 : array-like, shape=[n_samples, dim_2, dim_3]
+        ...
+        an : array-like, shape=[n_samples, dim_n-1, dim_n]
+
+        Returns
+        -------
+        mul : array-like, shape=[n_samples, dim_1, dim_n]
+        """
+        return reduce(gs.matmul, args)
```

Comment:
Just a comment: I think this can be made much faster using `gs.einsum`, because with such an implementation you don't have to incrementally multiply all the matrices together. On the other hand I don't know you would do this to support any number of factors ...

Reply:
Hmm I guess you mean something like `gs.einsum('...ij, ...jk, ...kl->...il', a, b, c)`

Reply:
I think it's good to have it that way for now because we know `gs.matmul` vectorizes correctly

Reply:
For three matrices this works, but your current implementation even supports n matrices :) I am fine with this, if we need to go fast we can always add a special case for three or four matrices later on.

Reply:
The 'any number of factors' is just to make explicit the associativity of `mul` 
It gets more convenient and readable when you do matrix conjugation for instance: 
```
point2 = Mat.mul(Mat.exp(vec), point, Mat.exp(-vec))
```

Reply:
Yes I totally agree with you :) I just wanted to make the point that `gs.einsum` might be faster in multiplying for example four matrices (because you can skip the intermediate results), but it is not clear (at least to me) how one would implement it for an arbitrary number of factors.

Thus I prefer the solution you commited right now.